Jenkinsfile

```diff
@@ -0,0 +1,178 @@
+        BAHTTP_PROXY = "${env.HTTP_PROXY ? '--build-arg HTTPS_PROXY="' + env.HTTPS_PROXY + '" --build-arg https_proxy="' + env.HTTPS_PROXY + '"' : ''}"
+        UID=sh(script: "id -u", returnStdout: true)
+        BUILDARGS = "--build-arg NOBUILD=1 --build-arg UID=$env.UID $env.BAHTTP_PROXY $env.BAHTTPS_PROXY"
+    }
+
+    options {
+        // preserve stashes so that jobs can be started at the test stage
+        preserveStashes(buildCount: 5)
+        timestamps ()
+    }
+
+    stages {
+        stage('Pre-build') {
+            parallel {
+                stage('checkpatch') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos:7'
+                            dir 'utils/docker'
+                            label 'docker_runner'
+                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS'
+                        }
+                    }
+                    steps {
+                        checkPatch user: GITHUB_USER_USR,
+                                   password: GITHUB_USER_PSW
+                    }
+                    post {
+                        always {
+                            archiveArtifacts artifacts: 'pylint.log', allowEmptyArchive: true
+                        }
+                    }
+                }
+            }
+        }
+        stage('Build') {
+            // abort other builds if/when one fails to avoid wasting time
+            // and resources
+            failFast true
+            parallel {
+                stage('Build on CentOS 7') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos:7'
+                            dir 'utils/docker'
+                            label 'docker_runner'
+                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS'
+                        }
+                    }
+                    steps {
+                        sconsBuild clean: "_build.external${arch}"
+                        stash name: 'CentOS-install', includes: 'install/**'
+                        stash name: 'CentOS-build-vars', includes: ".build_vars${arch}.*"
+                    }
+                    post {
+                        always {
+                            recordIssues enabledForFailure: true,
+                                         aggregatingResults: true,
+                                         id: "analysis-centos7",
+                                         tools: [ clang(), cppCheck() ],
+                                         filters: [excludeFile('.*\\/_build\\.external-Linux\\/.*'),
+                                                   excludeFile('_build\\.external-Linux\\/.*')]
+                        }
+                    }
```

Comment:
You need a `success { }` block here that removes `_build.external${arch}` for space management on Jenkins.  See the cart `Jenkinsfile`.